src/A2A/Server/ITaskStore.cs

```diff
@@ -9,44 +9,50 @@ public interface ITaskStore
     /// </summary>
     /// <param name="taskId">The ID of the task.</param>
     /// <param name="notificationConfigId">The ID of the push notification configuration.</param>
+    /// <param name="cancellationToken">A cancellation token that can be used to cancel the operation.</param>
     /// <returns>The push notification configuration if found, null otherwise.</returns>
-    Task<TaskPushNotificationConfig?> GetPushNotificationAsync(string taskId, string notificationConfigId);
+    Task<TaskPushNotificationConfig?> GetPushNotificationAsync(string taskId, string notificationConfigId, CancellationToken cancellationToken = default);
 
     /// <summary>
     /// Updates the status of a task.
     /// </summary>
     /// <param name="taskId">The ID of the task.</param>
     /// <param name="status">The new status.</param>
     /// <param name="message">Optional message associated with the status.</param>
+    /// <param name="cancellationToken">A cancellation token that can be used to cancel the operation.</param>
     /// <returns>The updated task status.</returns>
-    Task<AgentTaskStatus> UpdateStatusAsync(string taskId, TaskState status, Message? message = null);
+    Task<AgentTaskStatus> UpdateStatusAsync(string taskId, TaskState status, Message? message = null, CancellationToken cancellationToken = default);
 
     /// <summary>
     /// Stores or updates a task.
     /// </summary>
     /// <param name="task">The task to store.</param>
+    /// <param name="cancellationToken">A cancellation token that can be used to cancel the operation.</param>
     /// <returns>A task representing the operation.</returns>
-    Task SetTaskAsync(AgentTask task);
+    Task SetTaskAsync(AgentTask task, CancellationToken cancellationToken = default);
 
     /// <summary>
     /// Stores push notification configuration for a task.
     /// </summary>
     /// <param name="pushNotificationConfig">The push notification configuration.</param>
+    /// <param name="cancellationToken">A cancellation token that can be used to cancel the operation.</param>
     /// <returns>A task representing the operation.</returns>
-    Task SetPushNotificationConfigAsync(TaskPushNotificationConfig pushNotificationConfig);
+    Task SetPushNotificationConfigAsync(TaskPushNotificationConfig pushNotificationConfig, CancellationToken cancellationToken = default);
 
     /// <summary>
     /// Retrieves push notifications for a task.
     /// </summary>
     /// <param name="taskId">The ID of the task.</param>
+    /// <param name="cancellationToken">A cancellation token that can be used to cancel the operation.</param>
     /// <returns>A list of push notification configurations for the task.</returns>
-    Task<IEnumerable<TaskPushNotificationConfig>> GetPushNotificationsAsync(string taskId);
+    Task<IEnumerable<TaskPushNotificationConfig>> GetPushNotificationsAsync(string taskId, CancellationToken cancellationToken = default);
```

Comment:
Should/could this be 
```suggestion
    IAsyncEnumerable<TaskPushNotificationConfig> GetPushNotificationsAsync(string taskId, CancellationToken cancellationToken = default);
```
instead?

Reply:
I've created a dedicated issue for it: https://github.com/a2aproject/a2a-dotnet/issues/72 (I am going to mark it as resolved, so the PR can be merged)